js/index.js

```diff
@@ -44,26 +44,97 @@ function renderPepperoni() {
 
 function renderMushrooms() {
   // Iteration 1: set the visibility of `<section class="mushroom">`
+  document.querySelectorAll('.mushroom').forEach(oneMush => {
+    if (state.mushrooms) {
+      oneMush.style.visibility = 'visible';
+    } else {
+      oneMush.style.visibility = 'hidden';
+    }
+  });
 }
 
 function renderGreenPeppers() {
   // Iteration 1: set the visibility of `<section class="green-pepper">`
+  document.querySelectorAll('.green-pepper').forEach(oneGP => {
+    if (state.greenPeppers) {
+      oneGP.style.visibility = 'visible';
+    } else {
+      oneGP.style.visibility = 'hidden';
+    }
+  });
 }
 
+
 function renderWhiteSauce() {
   // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+  document.querySelectorAll('.sauce').forEach(oneSauce => {
+    if (state.whiteSauce) {
+      oneSauce.classList.add('sauce-white');
+    } else {
+      oneSauce.classList.remove('sauce-white');
+    }
+  });
 }
 
+// let changeCrust = document.getElementsByClassName("crust");
+// changeCrust[0].classList.remove("crust-gluten-free");
 function renderGlutenFreeCrust() {
   // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
+  document.querySelectorAll('.crust').forEach(oneCrust => {
+    if (state.glutenFreeCrust) {
+      oneCrust.classList.add("crust-gluten-free")
+    } else {
+      oneCrust.classList.remove("crust-gluten-free")
+    }
+  });
 }
 
 function renderButtons() {
   // Iteration 3: add/remove the class "active" of each `<button class="btn">`
+
+  // const pepperoniButtons = document.querySelectorAll('.btn.btn-pepperoni');
+  // const firstPepperoniButton = pepperoniButtons[0];
+  const firstPepperoniButton = document.querySelector('.btn.btn-pepperoni'); // == pepperoniButtons[0]
+  if (state.pepperoni) {
+    firstPepperoniButton.classList.add('active')
+  }
+  else {
+    firstPepperoniButton.classList.remove('active')
+  }
+  const firstMushroomButton = document.querySelector('.btn.btn-mushrooms'); 
+  if (state.mushrooms) {
+    firstMushroomButton.classList.add('active')
+  }
+  else {
+    firstMushroomButton.classList.remove('active');
+  }
+  const firstGreenPeppersButton = document.querySelector('.btn.btn-green-peppers'); 
+  if (state.greenPeppers) {
+    firstGreenPeppersButton.classList.add('active')
+  }
+  else {
+    firstGreenPeppersButton.classList.remove('active');
+  }
+  const firstWhiteSauceButton = document.querySelector('.btn.btn-sauce'); 
+  if (state.whiteSauce) {
+    firstWhiteSauceButton.classList.add('active')
+  }
+  else {
+    firstWhiteSauceButton.classList.remove('active');
+  }
+  const firstGlutenFreeButton = document.querySelector('.btn.btn-crust'); 
+  if (state.glutenFreeCrust) {
+    firstGlutenFreeButton.classList.add('active')
+  }
+  else {
+    firstGlutenFreeButton.classList.remove('active');
+  }
 }
```

Comment:
nice approach!
As an example, there is also this approach that is shorter and cleaner.
`
function renderButtons() {
  var ingArr = Object.keys(state);
  document.querySelectorAll(".btn").forEach(function(item, index) {
    if (state[ingArr[index]]) {
      item.classList.add("active");
    } else {
      item.classList.remove("active");
    }
  });
}
`